pymc/sampling/mcmc.py

```diff
@@ -729,8 +727,18 @@ def joined_blas_limiter():
         )
 
     if random_seed == -1:
+        warnings.warn(
+            "Setting random_seed = -1 is deprecated. Pass `None` to not specify a seed.",
+            FutureWarning,
+        )
         random_seed = None
-    rngs = get_random_generator(random_seed).spawn(chains)
+    elif isinstance(random_seed, tuple | list):
+        warnings.warn(
+            "A list or tuple of random_seed no longer specifies the specific random_seed of each chain. "
+            "Use a single seed instead.",
+            UserWarning,
+        )
+    rngs = np.random.SeedSequence(get_random_generator(random_seed)).spawn(chains)
```

Comment:
This needs to change a bit. `np.random.SeedSequence` will create a new `SeedSequence`, not a `Generator`. I suppose that you can access the generator's seed sequence as `get_random_generator(random_seed).bit_generator.seed_seq`, and then `spawn` from there.

From the numpy source, it looks rather easy to be able to [spawn new `Generator`s](https://github.com/numpy/numpy/blob/main/numpy/random/_generator.pyx#L301) from `BitGenerator`s. But the spawn method for those was also added on version 1.25, so we'll have to also look at how [`BitGenerator.spawn`](https://github.com/numpy/numpy/blob/main/numpy/random/bit_generator.pyx#L632) works off of seed sequences.

I think that the way to go would look something like:

```python
bit_gen = rng.bit_generator
seed_seq = bit_gen.seed_seq
rngs = [type(rng)(type(bit_gen)(seed)) for seed in seed_seq.spawn(chains)]
```

Reply:
This could be unpacked into a utility function and used on the other similar calls

Reply:
I see, so maybe lets just bumpy numpy min version?

Reply:
Okay numpy is dropping support for 1.24 on december 18, so maybe it's just fine: https://numpy.org/neps/nep-0029-deprecation_policy.html#drop-schedule

Reply:
I’m in favour of bumping